lib/mixins/dropdown.js

```diff
@@ -1,49 +1,50 @@
-import Popper from 'popper.js';
-
-import clickoutMixin from './clickout';
-import listenOnRootMixin from './listen-on-root';
-import { from as arrayFrom } from '../utils/array';
-import { assign } from '../utils/object';
-import warn from '../utils/warn';
+import Popper from "popper.js";
+import clickoutMixin from "./clickout";
+import listenOnRootMixin from "./listen-on-root";
+import { from as arrayFrom } from "../utils/array";
+import { assign } from "../utils/object";
 
 // Determine if an HTML element is visible - Faster than CSS check
-const isVisible = el => el && (el.offsetWidth > 0 || el.offsetHeight > 0);
+function isVisible(el) {
+    return el && (el.offsetWidth > 0 || el.offsetHeight > 0);
+}
 
 // Return an Array of visible items
-const filterVisible = els => (els ? els.filter(el => isVisible(el)) : []);
```

Comment:
Ah, this is what was causing the `this$1 = this` warning

Reply:
Yeah, the arrow functions are trying to capture `this`, which according to es 2015 spec at the top level of s module is undefined. Rollup has to do that manually to emulate the module environment accurately.

Reply:
Will know :)

Reply:
The warning is still issued, but I can't find any straggler arrow functions at top-level module scope. I'm thinking it might be in popper.js itself since the warning comes right where popper is imported.

Reply:
see lower comment )

Reply:
Just changed the `generateId` in tooltip.js class  back into a named function...

Reply:
![image](https://user-images.githubusercontent.com/16269252/29782118-2dc18370-8bd0-11e7-9ab7-c0b767852472.png)

Just pulled, and it's still warning. It's one of those things that _should be fine_, but can potentially introduce bugs.

Reply:
where did you get that warnings? i dont have any

Reply:
So it's not arrow functions gens this warning?

Reply:
![image](https://user-images.githubusercontent.com/20986566/29782636-86914b44-8c47-11e7-95a1-04514b70d9c4.png)
Also i have this error

Reply:
I get it running `yarn` which does the build after install.

I was asking Troy about that Popper error the other day too. It doesn't make much sense to me... The headless browser the jest test runner uses should have something like that in it's environment.

Reply:
https://stackoverflow.com/questions/42213522/mocking-document-createrange-for-jest

Reply:
I wonder if we can import popper at a different point, within the tooltip class? maybe in the constructor function, or win the show method (where be instantiate popper?  Or, if it is a global, then do we need to do the import at all?

Reply:
but source of warning is dropdown, not tooltip....

Reply:
or use a `const Popper = require('popper.js')` in the `constructor` or in the `show()` method?

Reply:
True...

Maybe in dropdown, we get Popper on create/show?